src/gui/layout/qgslayoutelevationprofilewidget.cpp

```diff
@@ -688,3 +716,33 @@ void QgsLayoutElevationProfileWidget::updateItemLayers()
   mProfile->setLayers( layers );
   mProfile->update();
 }
+
+void QgsLayoutElevationProfileWidget::layoutAtlasToggled( bool atlasEnabled )
+{
+  if ( atlasEnabled &&
+       mProfile && mProfile->layout() && mProfile->layout()->reportContext().layer()
+       && mProfile->layout()->reportContext().layer()->geometryType() == QgsWkbTypes::LineGeometry )
```

Comment:
@nyalldawson , would there be any point in supporting polygon geometry and use the exterior ring as curve? Probably not :)